src/main/kotlin/no/fdk/dataset_catalog/service/DatasetService.kt

```diff
@@ -31,46 +31,76 @@ class DatasetService(
     private var datasetUriPattern: Regex? = null
 
     fun getDatasetUriPattern(): Regex {
-        if(datasetUriPattern == null) {
+        if (datasetUriPattern == null) {
             datasetUriPattern = "^${applicationProperties.catalogUriHost}/\\d+/datasets/".toRegex()
         }
         return datasetUriPattern as Regex
     }
 
     fun getAll(catalogId: String, specializedTypeString: String? = null): List<Dataset> {
         val specializedType = specializedTypeFromString(specializedTypeString)
-
-        return if (specializedType == null) datasetRepository.findByCatalogId(catalogId) as List<Dataset>
+        val datasetList = if (specializedType == null) datasetRepository.findByCatalogId(catalogId) as List<Dataset>
         else datasetRepository.findByCatalogIdAndSpecializedType(catalogId, specializedType) as List<Dataset>
+
+        return datasetList.map { it.addOldAccessUrisToNewField() }
+    }
+
+
+    private fun Distribution.addOldAccessUrisToNewField(): Distribution {
+        val updatedAccessServiceUris: MutableSet<String> =
+            accessServiceUris?.toMutableSet()
+                ?: emptySet<String>().toMutableSet()
+        accessService?.mapNotNull { it.uri }
+            ?.forEach { updatedAccessServiceUris.add(it) }
+        return if (updatedAccessServiceUris.isNotEmpty()) {
+            copy(
+                accessServiceUris = updatedAccessServiceUris
+            )
+        } else this
```

Comment:
ikke noe du trenger å gjøre, føler bare at det blir litt mer ryddig
```suggestion
        return if (updatedAccessServiceUris.isEmpty()) this
            else copy(accessServiceUris = updatedAccessServiceUris)
```